### PR TITLE
feat(server): add `http2_max_concurrent_streams` builder option

### LINKED ISSUE #1772: Add builder option for HTTP2 max concurrent streams
The `SETTINGS_MAX_CONCURRENT_STREAMS` option controls how many concurrent streams a peer is willing to receive. Since we don't yet has push promises, this only makes sense in the server case. The corresponding method exists on the `h2` type.

- [ ] `server::Builder::http2_max_concurrent_streams`
- [ ] `server::conn::Http::http2_max_concurrent_streams`

### PR DESCRIPTION
**Pull request by seanmonstar**
Closes #1772

### CODE REVIEW

File: src/server/mod.rs [hunk 1]
```diff
@@ -302,6 +302,17 @@ impl<I, E> Builder<I, E> {
...
     }
 
+    /// Sets the [`SETTINGS_MAX_CONCURRENT_STREAMS`][spec] option for HTTP2
+    /// connections.
+    ///
+    /// Default is no limit (`None`).
+    ///
+    /// [spec]: https://http2.github.io/http2-spec/#SETTINGS_MAX_CONCURRENT_STREAMS
+    pub fn http2_max_concurrent_streams(mut self, max: impl Into<Option<u32>>) -> Self {
+        self.protocol.http2_max_concurrent_streams(max.into());
```
Comment (kleimkuhler) [inline src/server/mod.rs]: Why do you have `max.into()` here, but

```
if let Some(max) = max.into {
    ...
}
```

in `src/server/conn`?

Comment (seanmonstar) [reply src/server/mod.rs]: Hm, didn't think much about it, really!

Comment (kleimkuhler) [reply src/server/mod.rs]: Oh okay!

Comment (kleimkuhler) [reply src/server/mod.rs]: Oh no this makes sense. `h2`'s `fn max_concurrent_streams` needs a value, while here you are passing in an `Option` to `conn::Builder`'s wrapper.

**Pull request merged**